NEWS.md

```diff
@@ -0,0 +1,293 @@
+  ```sh
+  # true-ish last output value exits with zero
+  $ jq -ne true ; echo $?
+  true
+  0
+  # false-ish last output value (false and null) exits with 1
+  $ jq -ne false ; echo $?
+  false
+  1
+  # no output value exists with 4
+  $ jq -ne empty ; echo $?
+  4
+  ```
+- Add `--binary`/`-b` on Windows for binary output. To get `\n` instead of `\r\n` line endings. 0dab2b1 @nicowilliams
+- Add `--raw-output0` for NUL (zero byte) separated output. @asottile @pabs3 @itchyny #1990 #2235 #2684
+  ```sh
+  # will output a zero byte after each output
+  $ jq -n --raw-output0 '1,2,3' | xxd
+  00000000: 3100 3200 3300                           1.2.3.
+  # can be used with xargs -0
+  $ jq -n --raw-output0 '"a","b","c"' | xargs -0 -n1
+  a
+  b
+  c
+  $ jq -n --raw-output0 '"a b c", "d\ne\nf"' | xargs -0 printf '%q\n'
+  'a b c'
+  'd'$'\n''e'$'\n''f'
+  # can be used with read -d ''
+  $ while IFS= read -r -d '' json; do
+  >   jq '.name' <<< "$json"
+  > done < <(jq -n --raw-output0 '{name:"a b c"},{name:"d\ne\nf"}')
+  "a b c"
+  "d\ne\nf"
+  # also it's an error to output a string containing a NUL when using NUL separator
+  $ jq -n --raw-output0 '"\u0000"'
+  jq: error (at <unknown>): Cannot dump a string containing NUL with --raw-output0 option
+  ```
+- Fix assert crash and validate JSON for `--jsonarg`. @wader #2658
+- Remove deprecated `--argfile` option. @itchyny #2768
+
+## Language changes
+
+- Use decimal number literals to preserve precision. Comparison operations respects precision but arithmetic operations might truncate. #1752 @leonid-s-usov
+  ```sh
+  # precision is preserved
+  $ jq -n '100000000000000000'
+  100000000000000000
+  # comparison respects precision (this is false in JavaScript)
+  $ jq -n '100000000000000000 < 100000000000000001'
+  true
+  # arithmetic operations might truncate (same as JavaScript)
+  $ jq -n '100000000000000000+10'
+  100000000000000020
+  ```
+- Adds new builtin `pick(stream)` to emit a projection of the input object or array. @pkoppstein #2656
+  ```sh
+  $ jq -n '{"a": 1, "b": {"c": 2, "d": 3}, "e": 4} | pick(.a, .b.c, .x)'
+  {
+    "a": 1,
+    "b": {
+      "c": 2
+    },
+    "x": null
+  }
+  ```
+- Adds new builtin `debug(msgs)` that works like `debug` but applies a filter on the input before writing to stderr.
+  ```sh
+  $ jq -n '1 as $x | 2 | debug("Entering function foo with $x == \($x)", .) | (.+1)'
+  ["DEBUG:","Entering function foo with $x == 1"]
+  ["DEBUG:",2]
+  3
+  $ jq -n '{a: 1, b: 2, c: 3} | debug({a, b, sum: (.a+.b)})'
+  ["DEBUG:",{"a":1,"b":2,"sum":3}]
+  {
+    "a": 1,
+    "b": 2,
+    "c": 3
+  }
+  ```
+- Adds new builtin `scan($re; $flags)`. Was documented but not implemented. @itchyny #1961
+  ```sh
+  # look for pattern "ab" in "abAB" ignoring casing
+  $ jq -n '"abAB" | scan("ab"; "i")'
+  "ab"
+  "AB"
+  ```
+- Adds new builtin `abs` to get absolute value. This was previously possibly using `length` or `fabs` but naming was a bit confusing. @pkoppstein #2767
```

Comment:
```suggestion
- Adds new builtin [`abs`](https://jqlang.github.io/jq/manual/#abs) to get absolute value. This was previously possibly using `length` or `fabs` but naming was a bit confusing. @pkoppstein #2767
```

Reply:
@nicowilliams @itchyny  what do you think? links?